### PR TITLE
Update azure_rm_deployment related document

### PR DESCRIPTION
**Pull request by Fred-sun**
##### SUMMARY

Update azure_rm_deployment related document.

##### ISSUE TYPE

 
- Docs Pull Request

##### COMPONENT NAME

azure_rm_deployment/azure_rm_deployment_facts
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by Fred-sun** [comment]
ready_for_review

File: lib/ansible/modules/cloud/azure/azure_rm_deployment.py [hunk 1]
```diff
@@ -21,67 +21,68 @@
...
-        Either one of them is required if "state" parameter is "present".
+        - A hash containing the templates inline. This parameter is mutually exclusive with I(template_link).
+        - Either one of them is required if I(state=present).
     type: dict
   template_link:
     description:
-      - Uri of file containing the template body. This parameter is mutually exclusive with 'template'. Either one
-        of them is required if "state" parameter is "present".
+        - Uri of file containing the template body. This parameter is mutually exclusive with I(template).
+        - Either one of them is required if I(state=present).
```
Comment (acozine) [inline lib/ansible/modules/cloud/azure/azure_rm_deployment.py]:
```suggestion
        - Either I(template) or I(template_link) is required if I(state=present).
```

File: lib/ansible/modules/cloud/azure/azure_rm_deployment.py [hunk 2]
```diff
@@ -21,67 +21,68 @@
...
     default: incremental
     choices:
         - complete
         - incremental
   template:
     description:
-      - A hash containing the templates inline. This parameter is mutually exclusive with 'template_link'.
-        Either one of them is required if "state" parameter is "present".
+        - A hash containing the templates inline. This parameter is mutually exclusive with I(template_link).
+        - Either one of them is required if I(state=present).
```
Comment (acozine) [inline lib/ansible/modules/cloud/azure/azure_rm_deployment.py]:
```suggestion
        - Either I(template_link) or I(template) is required if I(state=present).
```

File: lib/ansible/modules/cloud/azure/azure_rm_deployment.py [hunk 3]
```diff
@@ -21,67 +21,68 @@
...
-      - Uri of file containing the template body. This parameter is mutually exclusive with 'template'. Either one
-        of them is required if "state" parameter is "present".
+        - Uri of file containing the template body. This parameter is mutually exclusive with I(template).
+        - Either one of them is required if I(state=present).
   parameters:
     description:
-      - A hash of all the required template variables for the deployment template. This parameter is mutually exclusive
-        with 'parameters_link'. Either one of them is required if "state" parameter is "present".
+        - A hash of all the required template variables for the deployment template. This parameter is mutually exclusive with I(parameters_link).
+        - Either one of them is required if I(state=present).
```
Comment (acozine) [inline lib/ansible/modules/cloud/azure/azure_rm_deployment.py]:
```suggestion
        - Either I(parameters) or I(parameters_links) is required if I(state=present).
```

File: lib/ansible/modules/cloud/azure/azure_rm_deployment.py [hunk 4]
```diff
@@ -21,67 +21,68 @@
...
     type: dict
   parameters_link:
     description:
-      - Uri of file containing the parameters body. This parameter is mutually exclusive with 'parameters'. Either
-        one of them is required if "state" parameter is "present".
+        - Uri of file containing the parameters body. This parameter is mutually exclusive with I(parameters).
+        - Either one of them is required if I(state=present).
```
Comment (acozine) [inline lib/ansible/modules/cloud/azure/azure_rm_deployment.py]:
```suggestion
        - Either I(parameters) or I(parameters_links) is required if I(state=present).
```

**Review by acozine — COMMENTED** [review]
Overall this is an improvement. I made a few suggestions to make it even better.

**Comment by Fred-sun** [comment]
ready_for_review

**Comment by acozine** [comment]
Sorry @Fred-sun, I introduced a spelling mistake in my suggestions. I fixed it and will merge this PR when it passes our CI system.

**Pull request merged**